mailet/crypto/src/main/java/org/apache/james/transport/KeyStoreHolderConfiguration.java

```diff
@@ -0,0 +1,123 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport;
+
+import java.security.KeyStore;
+import java.util.Optional;
+
+import org.apache.mailet.MailetConfig;
+
+public class KeyStoreHolderConfiguration {
+    public static final String FILE_TYPE = "fileType";
+    public static final String KEY_STORE_TYPE = "keyStoreType";
+    public static final String KEY_STORE_FILE_NAME = "keyStoreFileName";
+    public static final String KEY_STORE_PASSWORD = "keyStorePassword";
+    public static final String PEM_FILE_NAME = "pemFileName";
+    public static final String KEY_STORE_TYPE_DEFAULT_VALUE = KeyStore.getDefaultType();
+    public static final String KEY_STORE_PASSWORD_DEFAULT_VALUE = "";
+
+    public static class Builder {
+        private Optional<String> fileType;
+        private Optional<String> keyStoreType;
+        private Optional<String> keyStoreFileName;
+        private Optional<String> keyStorePassword;
+        private Optional<String> pemFileName;
+
+        public Builder setFileType(Optional<String> fileType) {
+            this.fileType = fileType;
+            return this;
+        }
+
+        public Builder setKeyStoreType(Optional<String> keyStoreType) {
+            this.keyStoreType = keyStoreType;
+            return this;
+        }
+
+        public Builder setKeyStoreFileName(Optional<String> keyStoreFileName) {
+            this.keyStoreFileName = keyStoreFileName;
+            return this;
+        }
+
+        public Builder setKeyStorePassword(Optional<String> keyStorePassword) {
+            this.keyStorePassword = keyStorePassword;
+            return this;
+        }
+
+        public Builder setPemFileName(Optional<String> pemFileName) {
+            this.pemFileName = pemFileName;
+            return this;
+        }
+
+        public KeyStoreHolderConfiguration build() {
+            return new KeyStoreHolderConfiguration(this.fileType.map(type -> KeyFileType.valueOf(type.toUpperCase())).orElse(KeyFileType.KEYSTORE),
```

Comment:
Move `this.fileType.map(type -> KeyFileType.valueOf(type.toUpperCase())).orElse(KeyFileType.KEYSTORE)` into a `KeyFileType::parse` method...